### DISCUSSION TITLE
Laptop recommendations for MONAILabel?

### DISCUSSION THREAD

**Discussion by lassoan** [Project-MONAI/MONAILabel#833] [General]
@SachidanandAlle made a comment in another thread: "You can even run it on your laptop (if you have some NVIDIA GPU)"

I'm wondering what laptop and GPU would you recommend for MONAILabel?

Most laptops use RTX 3060/3070/3080 with 6/8/16GB RAM - are these sufficient for MONAILabel? Our impression is that to comfortably run most MONAILabel demo applications, we need 24GB GPU RAM.

GPU laptops are usually huge, very heavy (4-6 kg), battery life is terrible (few hours), they are noisy, and overheating. Has anyone found a laptop that has a strong GPU but still somewhat portable?

**Comment by SachidanandAlle** [comment]
you can run basic spleen segmentation app with 6GB to 8GB RAM.. this is useful only when you don't have good server with GPU..
Many times people like to explore what is MONAILabel using 3D slicer... many times people want to start with basic inference or small training on a sample image.. to see how it works.. in such cases (for demo kind of situation), users can simply install both server and slicer on their laptop and browse through some basic features of MONAILabel without need of a heavy server

I could do basic testing for monailabel (segmentation spleen or pathology models) on my windows laptop with following details:

```
PS C:\> nvidia-smi
Sat Jun 11 15:21:14 2022
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 511.69       Driver Version: 511.69       CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Quadro RTX 3000... WDDM  | 00000000:01:00.0 Off |                  N/A |
| N/A   47C    P8     7W /  N/A |      0MiB /  6144MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

**Reply by lassoan** [reply]
Thank you. It would be nice to specify the hardware requirements for MONAILabel apps so that people know which ones are expected to work on their computer.

**Reply by diazandr3s** [reply]
Agree! A benchmark carried out on different GPUs and image sizes or ROI sizes will help.

**Comment by diazandr3s** [comment]
This is a good question, @lassoan.

As an example, there is a user working with MONAI Label on a 4GB memory GPU. [Here](https://discourse.slicer.org/t/how-to-start-with-monailabel-for-new-models/21063/70?u=diazandr3s) you can see the conversation about that.

Performance depends on which model you use to annotate images. [DeepEdit](https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology#deepedit) work on whole volumes and the [Segmentation](https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology#segmentation) model works on patches. Depending on the image size (DeepEdit) or patch size (Segmentation) you defined and the GPU size, you should be able to train and perform inference on images.

Hope that helps.
